src/lib/textToCad.ts

```diff
@@ -68,29 +65,25 @@ async function getTextToCadResult(
   return data
 }
 
-interface TextToKclProps {
+interface TextToKclPropsApplicationLevel {
   trimmedPrompt: string
-  fileMachineSend: (
-    type: EventFrom<typeof fileMachine>,
-    data?: unknown
-  ) => unknown
   navigate: NavigateFunction
-  context: ContextFrom<typeof fileMachine>
   token?: string
-  settings: {
-    theme: Themes
+  projectName: string
+  isProjectNew: boolean
+  settings?: {
     highlightEdges: boolean
   }
 }
 
-export async function submitAndAwaitTextToKcl({
+export async function submitAndAwaitTextToKclSystemIO({
   trimmedPrompt,
-  fileMachineSend,
-  navigate,
-  context,
   token,
+  projectName,
+  navigate,
+  isProjectNew,
   settings,
-}: TextToKclProps) {
+}: TextToKclPropsApplicationLevel) {
```

Comment:
Nit: I think this type name and `submitAndAwaitTextToKclSystemIO` are left over from when you were gonna leave the within-file Text-to-CAD workflow as-is